### PR TITLE
Fix bitwise decimal & octal

### LINKED ISSUE #2450: Error in userSetup.sh
### Is there a pinned issue for this?

- [X] I have read the pinned issues and could not find my issue

### Is there an existing or similar issue/discussion for this?

- [X] I have searched the existing issues
- [X] I have searched the existing discussions

### Is there any comment in the documentation for this?

- [X] I have read the documentation, especially the FAQ and Troubleshooting parts

### Is this related to a provider?

- [X] I have checked the provider repo for issues
- [X] My issue is NOT related to a provider

### Are you using the latest release?

- [X] I am using the latest release

### Have you tried using the dev branch latest?

- [ ] I have tried using dev branch

### Docker run config used

```
version: '3.3'
services:
    transmission-openvpn:
        cap_add:
            - NET_ADMIN
        volumes:
            - '/zpool1/appstorage/transmission:/data'
            - '/zpool1/media/movies:/movies'
            - '/zpool1/media/tv:/tv'
            - '/zpool1/media/music:/music'
        environment:
            - OPENVPN_PROVIDER=***
            - OPENVPN_CONFIG=***
            - OPENVPN_USERNAME=***
            - OPENVPN_PASSWORD=***
            - LOCAL_NETWORK=***
            - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
            - PUID=997 # transmission-user
            - PGID=997 # nas.mediagroup
            - TRANSMISSION_UMASK=504 # decimal of 0770
            - TRANSMISSION_DOWNLOAD_DIR=/data/unsorted
            - TRANSMISSION_INCOMPLETE_DIR=/data/incomplete
        logging:
            driver: json-file
            options:
                max-size: 10m
        ports:
            - "9091:9091"
        image: haugene/transmission-openvpn:4.0
        restart: unless-stopped
```

### Current Behavior

Data in `data/incomplete` is chmod to `-wxrwxr-x` aka `375` whenever the container (re)starts.  `264` for files.  This causes downloads to fail with permission errors (user has no read permission on these folders now!).

### Expected Behavior

Data in `data/incomplete` should chmod to `rwxtwx---` aka `770` whenever the container (re)starts.

### How have you tried to solve the problem?

Can be resolved by chmod incomplete to 770.  See below for suggested fix in userSetup.sh.

### Log output

You can see where it does this here:

```
transmission-transmission-openvpn-1  | Enforcing ownership on transmission config directories
transmission-transmission-openvpn-1  | Applying permissions to transmission config directories
transmission-transmission-openvpn-1  | Setting owner for transmission paths to 997:997
transmission-transmission-openvpn-1  | Setting permissions for download and incomplete directories
transmission-transmission-openvpn-1  | Mask: 770
transmission-transmission-openvpn-1  | Directories: 375
transmission-transmission-openvpn-1  | Files: 264
transmission-transmission-openvpn-1  | Setting permission for watch directory (775) and its files (664)
transmission-transmission-openvpn-1  |
transmission-transmission-openvpn-1  | -------------------------------------
transmission-transmission-openvpn-1  | Transmission will run as
transmission-transmission-openvpn-1  | -------------------------------------
transmission-transmission-openvpn-1  | User name:   abc
transmission-transmission-openvpn-1  | User uid:    997
transmission-transmission-openvpn-1  | User gid:    997
transmission-transmission-openvpn-1  | -------------------------------------
transmission-transmission-openvpn-1  |
transmission-transmission-openvpn-1  | STARTING TRANSMISSION
```

### HW/SW Environment

```markdown
- OS: Ubuntu 22.04
- Docker: Docker version 20.10.18, build b40c2f6
```

### Anything else?

Possibly related to: https://github.com/haugene/docker-transmission-openvpn/discussions/2278

There is an error in the bitwise & here? https://github.com/haugene/docker-transmission-openvpn/blob/b33d0fe4c938259a0d4eb844e55468f387456121/transmission/userSetup.sh#L56-L67

It seems likely that the intention for directories is octal 777 & octal 770 - but this snippet instead combines incompatible decimal and octal interpretations of the integer and uses an unexpected bitwise ~.

Since the user has input in decimal the easiest solution would be to refactor the masks to decimal: octal 777 -> decimal 511 and octal 666 -> decimal 438.   I have also removed the ~.

for folders:
```
DIR_PERMS=$(printf '%o\n' $((511 & TRANSMISSION_UMASK)))
```
and for files:
```
FILE_PERMS=$(printf '%o\n' $((438 & TRANSMISSION_UMASK)))
```
where `TRANSMISSION_UMASK` is the decimal umask provided as an environment variable (or extracted from the settings in *decimal* rather than octal).

I'd recommend replacing the inline with the variable at lines 65 and 67, eg: `-exec chmod $DIR_PERMS` and `-exec chmod $FILE_PERMS`

Thanks for taking a look!
J.P.

### PR DESCRIPTION
**Pull request by JPHutchins**
## Breaking change

```txt
<placeholder>
```

## Proposed change

```txt
Fix bitwise decimal & octal
```

## Type of change

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to this container)
- [ ] Breaking change (fix/feature causing existing functionality to break)

## Additional information

Also uses the TRASNMSISSION_UMASK variable instead of settings.json since updateSettings.py is called after userSetup.sh causing the TRANSMISSION_UMASK in settings.json to be stale when userSetup.sh accesses it.

- This PR fixes or closes issue: fixes #2450 
- This PR is related to issue: ```relates to #```
- Link to documentation updated (if done separately): ```https://...```

## Checklist

- [ ] The code change is tested and works locally.
- [x] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated

### CODE REVIEW

**Comment by pkishino** [comment]
please rebase to :dev branch. thanks

**Comment by JPHutchins** [comment]
@pkishino Should the PR be into dev instead of master then?

**Comment by pkishino** [comment]
Yes please

File: docs/supported-providers.md [hunk 1]
```diff
@@ -130,12 +130,12 @@ Compose sample:
             - OPENVPN_PASSWORD=pass
 ```
 
-### If you only need to mount one file
+### Do not mount single config file
 
-You might not need to mount a folder of configs. You may just have one config file you want to use.
-In that case you can just mount it directly. Mounting it as `default.ovpn` will let you omit `OPENVPN_CONFIG` as well.
+Do not mount a single config directly. The container will fail if you try, since it causes sed errors when modify-openvpn-config.sh is executed.
+Instead mount the directory where the config exists.
```
Comment (pkishino) [inline docs/supported-providers.md:136]: I think it is a good idea to add a comment like this but one can also disable the config modifications like I have using : MODIFY_CHOSEN_CONFIG and set it to false..

**Pull request merged**